### PR TITLE
IFP256

### LINKED ISSUE #114: `iFP256`
### Motivation

Fixed point integers currently do not have a signed 256-bit integer type.

### PR DESCRIPTION
**Pull request by rostyslavtyshko**
## Type of change

- New feature
Added `IFP256` - 256-bit signed fixed point number

## Changes

The following changes have been made:

-Added `IFP256` - 256-bit signed fixed point number

## Notes

Closes #114 

## Related Issues

Closes #\<issue number\>

### CODE REVIEW

**Review by nfurfaro — DISMISSED** [review]

File: libs/fixed_point/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 # Overview
 
-The Fixed Point Number library provides a library to use fixed-point numbers in Sway. It has 3 distinct unsigned types: `UFP32`, `UFP64` and `UFP128` as well as a signed type - `IFP64`. This type is stack allocated.
+The Fixed Point Number library provides a library to use fixed-point numbers in Sway. It has 3 distinct unsigned types: `UFP32`, `UFP64` and `UFP128` as well as 2 signed types `IFP64` and `IFP256`. These types are stack allocated.
```
Comment (bitzoic) [inline libs/fixed_point/README.md:10]: https://github.com/FuelLabs/sway-libs/pull/105 will cause merge conflicts with this. Will wait to review this until it's merged first

Comment (bitzoic) [reply libs/fixed_point/README.md:10]: This needs to be updated before we can merge

File: libs/fixed_point/src/ifp256.sw [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+            underlying: UFP128::min(),
+            non_negative: false,
+        }
+    }
+
+    pub fn zero() -> Self {
+        Self::from(UFP128::zero())
+    }
+
+    fn sign_reverse(self) -> Self {
```
Comment (bitzoic) [inline libs/fixed_point/src/ifp256.sw:48]: I think the same comments from the last PR apply to this one

**Review by bitzoic — CHANGES_REQUESTED** [review]

File: libs/fixed_point/src/ifp256.sw (hunk 2 above)
Comment (rostyslavtyshko) [reply libs/fixed_point/src/ifp256.sw]: Fixed

File: libs/fixed_point/src/ifp256.sw [hunk 3]
```diff
@@ -0,0 +1,306 @@
+library ifp256;
+// A wrapper library around the  type for mathematical functions operating with signed 256-bit fixed point numbers.
```
[the remark this answers is not in the document]
Comment (rostyslavtyshko) [reply libs/fixed_point/src/ifp256.sw]: Fixed

**Pull request merged**